downloads/downloads-impl/src/main/java/com/duckduckgo/downloads/impl/AndroidFileDownloader.kt

```diff
@@ -17,29 +17,43 @@
 package com.duckduckgo.downloads.impl
 
 import android.webkit.URLUtil
-import androidx.annotation.WorkerThread
-import com.duckduckgo.downloads.api.DownloadCallback
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.app.global.DispatcherProvider
 import com.duckduckgo.downloads.api.DownloadFailReason
 import com.duckduckgo.downloads.api.FileDownloader
 import com.duckduckgo.downloads.api.FileDownloader.PendingFileDownload
-import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
-class AndroidFileDownloader @Inject constructor(
+class AndroidFileDownloader constructor(
     private val dataUriDownloader: DataUriDownloader,
-    private val networkFileDownloader: NetworkFileDownloader
+    private val callback: FileDownloadCallback,
+    private val workManager: WorkManager,
+    @AppCoroutineScope private val coroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
 ) : FileDownloader {
 
-    @WorkerThread
-    override fun download(
+    override fun enqueueDownload(
         pending: PendingFileDownload,
-        callback: DownloadCallback
     ) {
         when {
-            pending.isNetworkUrl -> networkFileDownloader.download(pending, callback)
-            pending.isDataUrl -> dataUriDownloader.download(pending, callback)
+            pending.isNetworkUrl -> enqueueToWorker(pending)
+            // We don't delegate the data URLs to the worker because you can't pass a lot of data through the [Data] class.
```

Comment:
The data URLs have the file in the URL itself, eg. when long clicking on a image, the URL already contains the image.
Passing that URL through the `Data` structure to the worker will cause an exception as it can be more than 10k (limit of the `Data` struct)